drivers/mem_manage/mm_drv_common.c

```diff
@@ -0,0 +1,409 @@
+/*
+ * Copyright (c) 2021 Intel Corporation
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/**
+ * @file
+ * @brief Common Memory Management Driver Code
+ *
+ * This file provides common implementation of memory management driver
+ * functions, for example, sys_mm_drv_map_region() can use
+ * sys_mm_drv_map_page() to map page by page for the whole region.
+ * This avoids duplicate implementations of same functionality in
+ * different drivers. The implementations here are marked as
+ * weak functions so they can be overridden by the driver.
+ */
+
+#include <kernel.h>
+#include <string.h>
+#include <toolchain.h>
+#include <sys/check.h>
+#include <sys/util.h>
+
+#include <drivers/mem_manage/system_mem_manage.h>
+
+#include "mm_drv_common.h"
+
+struct k_spinlock sys_mm_drv_common_lock;
+
+bool sys_mm_drv_is_addr_array_aligned(uintptr_t *addr, size_t cnt)
+{
+	size_t idx;
+	bool ret = true;
+
+	for (idx = 0; idx < cnt; idx++) {
+		if (!sys_mm_drv_is_addr_aligned(addr[idx])) {
+			ret = false;
+			break;
```

Comment:
just `return false` and you can remove the `ret` variable.

Reply:
MISRA wants only one return statement per function... (errr...)